src/gateway/sharding/shard_runner.rs

```diff
@@ -248,22 +253,6 @@ impl ShardRunner {
                 }))
                 .await,
         );
-
-        // In return, we wait for either a Close Frame response, or an error, after which this WS
-        // is deemed disconnected from Discord.
-        loop {
-            match self.shard.client.next().await {
-                Some(Ok(tungstenite::Message::Close(_))) => return,
-                Some(Err(_)) => {
-                    warn!(
-                        "[ShardRunner {:?}] Received an error awaiting close frame",
-                        self.shard.shard_info(),
-                    );
-                    return;
-                },
-                _ => {},
-            }
-        }
```

Comment:
Do you think there might end up being a race condition during a restart where a reidentify is attempted before the close frame is received, and the reidentify fails as a result? That wouldn't be catastrophic as the shard would get re-queued and maybe a 5 second penalty would be incurred. However, maybe this could be controlled with a boolean parameter that indicates whether to wait for a close frame or not. Then, in the case where a resume fails, we can decide not to wait for the close frame since the behavior of Discord's gateway would be in question.

Reply:
I don't see why the reidentify could possibly fail due to a race condition with the "poisoned" websocket. The reidentify would be using an entirely new `Shard` with an entirely separate websocket connection.